MBBSEmu/CPU/CPUCore.cs

```diff
@@ -3651,6 +3660,120 @@ private void Op_Movsw()
 
         }
 
+        /// <summary>
+        ///     Move with Sign-Extend
+        /// </summary>
+        [MethodImpl(CompilerOptimizations)]
+        private void Op_Movsx()
+        {
+
+            var result = _currentOperationSize switch
+            {
+                2 => Op_Movsx_16(),
+                4 => Op_Movsx_32(),
+                _ => throw new Exception("Unsupported Operation Size")
+            };
+            
+            WriteToDestination(result);
+        }
+
+        /// <summary>
+        ///     Move with Sign-Extend to 16bit destination
+        /// </summary>
+        [MethodImpl(CompilerOptimizations)]
+        private ushort Op_Movsx_16()
+        {
+            var result = GetOperandValueUInt8(_currentInstruction.Op1Kind, EnumOperandType.Source).ToUshortSignExtended();
+            return result;
+        }
+
+        /// <summary>
+        ///     Move with Sign-Extend to 32bit destination
+        /// </summary>
+        [MethodImpl(CompilerOptimizations)]
+        private uint Op_Movsx_32()
+        {
+            var sourceSize = 0;
+
+            if (_currentInstruction.Op1Kind == OpKind.Register)
```

Comment:
Could these be done using a switch expression like the result below?

`var sourceSize = _currentInstruction.Op1Kind switch`

Reply:
yup! cleaned it up on both sx/zx and switched to 1 = 8bit, 2 = 16bit for consistency